spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/ReplicaPartitioner.scala

```diff
@@ -0,0 +1,34 @@
+package com.datastax.spark.connector.rdd.partitioner
+
+import java.net.InetAddress
+
+import com.datastax.spark.connector.cql.CassandraConnector
+import org.apache.spark.{Partition, Partitioner}
+
+
+case class EndpointPartition(index: Int, endpoint: Set[InetAddress]) extends Partition
```

Comment:
Would you make a trait `ICassandraPartition` or something which would be extended by `CassandraPartition` and `EndpointPartition`? they both share `index` and `endpoint` fields